### PR TITLE
Hack stack.sh failure (EUID=0); Update stackrc

### PR DESCRIPTION
**Pull request by noelmcloughlin**
PR includes following:

1. Fix `stack.sh` failure with `$EUID=0` despite salt `runas=stack`. Setenv `EUID=$(id -u stack)`.
```
                  + [[ 0 -eq 0 ]]
                  + set +o xtrace
              stdout:
                  DevStack should be run as a user with sudo permissions, 
                  not root
```
2. Bunch of other updates for devstack config.

### CODE REVIEW

File: devstack/init.sls [hunk 1]
```diff
@@ -58,23 +58,25 @@ openstack-devstack config:
...
-        rabbit_password: {{ devstack.conf.rabbit_password }}
         enable_httpd_mod_wsgi_services: {{ devstack.conf.enable_httpd_mod_wsgi_services }}
         keystone_use_mod_wsgi: {{ devstack.conf.keystone_use_mod_wsgi }}
         logfile: {{ devstack.dir.log }}/{{ devstack.conf.logfile }}
+        logdays: {{ devstack.conf.logdays }}
         verbose: {{ devstack.conf.verbose }}
         enable_debug_log_level: {{ devstack.conf.enable_debug_log_level }}
         enable_verbose_log_level: {{ devstack.conf.enable_verbose_log_level }}
         reclone: {{ devstack.conf.reclone }}
+        odl_mgr_ip: {{ devstack.conf.odl_mgr_ip }}
```
Comment (javierbertoli) [inline devstack/init.sls]: @noelmcloughlin, any particular reason to pass all the parameters individually, and not a single line like

```yaml
  - context:
      conf: {{ devstack.conf }}
```
and then, in [the jinja template](https://github.com/saltstack-formulas/devstack-formula/blob/master/devstack/files/localrc.j2) use `{{ conf.parameter }}`?

Comment (noelmcloughlin) [reply devstack/init.sls]: Computers are hard!!

**Comment by noelmcloughlin** [comment]
Thanks @javierbertoli  for review.    Some new issue but `stack.sh` is happier now.

```
                  ++ tools/install_prereqs.sh:source:95       :   date
                  + stack.sh:main:759                        :   [[ False != \T\r\u\e ]]
                  + stack.sh:main:760                        :   PYPI_ALTERNATIVE_URL=
                  + stack.sh:main:760                        :   /home/stack/tools/install_pip.sh
                  /home/stack/localrc: line 9: selenium-node: command not found
                  ++ stack.sh:main:760                        :   err_trap
                  ++ stack.sh:err_trap:556                    :   local r=127
                  stack.sh failed: full log in stack.sh.log.2018-08-25-180437
                  Error on exit

Summary for local
-------------
Succeeded: 13 (changed=8)
Failed:     1
-------------
Total states run:     14
Total run time:  111.215 s

```

File: devstack/init.sls (hunk 1 above)
Comment (javierbertoli) [reply devstack/init.sls]: :smile:

**Comment by javierbertoli** [comment]
```
/home/stack/localrc: line 9: selenium-node: command not found
                  ++ stack.sh:main:760                        :   err_trap
                  ++ stack.sh:err_trap:556                    :   local r=127
                  stack.sh failed: full log in stack.sh.log.2018-08-25-180437
                  Error on exit
```

is that a missing dependency, perhaps?

**Comment by noelmcloughlin** [comment]
I fixed `HOST_NAME` variable now (removed trailing white space) .  

But `HOST_IP` and `HOST_IPV6` still have garbled value (None) in `localrc.j2`. Any ideas?
```
HOST_IP=None
HOST_NAME=selenium-node
SERVICE_HOST=None
```
stack.sh eventually throws segmentation fault.
```
+ stack.sh:main:472                        :   exec
+ stack.sh:main:473                        :   [[ True == \T\r\u\e ]]
+ stack.sh:main:474                        :   _of_args=-v
+ stack.sh:main:475                        :   [[ True == \T\r\u\e ]]
+ stack.sh:main:476                        :   _of_args='-v --no-timestamp'
+ stack.sh:main:479                        :   exec
++ stack.sh:main:479                        :   /home/stack/tools/outfilter.py -v --no-timestamp -o stack.sh.log.2018-08-25-182058
[ERROR   ] retcode: 2
```

**Comment by noelmcloughlin** [comment]
Nevermind - fixed bug in `localrc.j2`.

**Comment by noelmcloughlin** [comment]
Hi @javierbertoli  could you approve/merge if happy with this.

Devstack appears to be very sensitive to environment and generated `local.con` configuration for stack.sh user.  I had many problems with `sudo` issue due to issue caused by value "root" (`salt.file.managed.local.conf['user']`) being passed into the context dictionary instead of "stack" (`salt.file.managed.local.conf['context']['user']`).  This was weird and maybe salt bug but I restructured the pillar data once I saw the problem.

Devstack supports various configurations so maybe my preferred config  ("mysql, keystone") has some known issue.  I have opened  #2 and #3 as known issues on Ubuntu/RedHat.

Cheers.

**Comment by javierbertoli** [comment]
Great work, @noelmcloughlin

**Comment by noelmcloughlin** [comment]
Thanks @javierbertoli appreciate your review.

**Comment by javierbertoli** [comment]
You're welcome, @noelmcloughlin ! :wink:

**Pull request merged**